lib/linux_stats/os/proc_mounts.rb

```diff
@@ -69,29 +102,72 @@ def report
       storage_report
     end
 
+    def strip_container_directory(mount_path = nil)
+      mount_path.sub! "/#{@container_prefix}",'/'
+      mount_path.sub! '//','/'
+    end
+
+    def add_container_directory(mount_path = nil)
+      mount_path.sub! /^\//,"/#{@container_prefix}/"
+      mount_path.sub! '//','/'
+    end
+
     # see https://www.ruby-forum.com/topic/4416522
     # returns: array of partition use: [
     #   <used kilobytes>
     #   <max kilobytes>
     # ]
     def partition_used(partition)
+      # Return magic number if in test_mode to prevent syscall
+      return '128' if @test_mode
       b = ' ' * 128
       syscall(137, partition, b)
       a = b.unpack('QQQQQ')
       [a[2] * blocks_per_kilobyte, a[4] * blocks_per_kilobyte]
     end
 
-    def mounts
+    def mounts(data = nil)
       mount_list = []
-      IO.readlines(@proc_data_source).each do |line|
-        mount = line.split[1]
-        next if IGNORE_PARTITIONS.include? mount
-        mount_list.push line.split[1].strip
-      end
-      IGNORE_PARTITIONS.each do |partition|
-        mount_list.reject! { |x| x =~ /#{partition}/ }
+      unless data
+        IO.readlines(@proc_data_source).each do |line|
+          mount = line.split[1]
+
+          # Inside a container, we should exclude everything not in the well-known host filesystem
+          # mount.
+          if @proc_data_source.include? 'hostproc'
+            next unless mount.include? @container_prefix
+          end
+
+          next if IGNORE_PARTITIONS.include? mount
+          mount_list.push line.split[1].strip
+        end
+      else
+        mount_test_data = StringIO.new(data)
+        mount_test_data.readlines.each do |line|
```

Comment:
This block and the previous one look like they're the same code - just a different way to get the input. It would be better to separate out the data collection pieces and have a single iteration through the mount data to prevent code duplication.